### PR TITLE
Make interop work.

### PR DESCRIPTION
**Pull request by duesee**
This is the final PR that makes interop (modulo some unsupported features) work.

### CODE REVIEW

File: openmls/src/group/core_group/mod.rs [hunk 1]
```diff
@@ -5,42 +5,6 @@
 //! This means that some functions that are not expected to fail and throw an
 //! error, will still return a `Result` since they may throw a `LibraryError`.
 
-// Private
```
Comment (duesee) [inline openmls/src/group/core_group/mod.rs:8]: This is only a reformat + an addition of `use log::`.

File: openmls/src/group/public_group/mod.rs [hunk 2]
```diff
@@ -79,7 +79,20 @@ impl PublicGroup {
         group_context: GroupContext,
         initial_confirmation_tag: ConfirmationTag,
     ) -> Self {
-        let interim_transcript_hash = vec![];
+        let backend = OpenMlsRustCrypto::default();
```
Comment (duesee) [inline openmls/src/group/public_group/mod.rs:82]: This is the important change.

File: openmls/src/group/public_group/mod.rs [hunk 3]
```diff
@@ -79,7 +79,20 @@ impl PublicGroup {
         group_context: GroupContext,
         initial_confirmation_tag: ConfirmationTag,
     ) -> Self {
-        let interim_transcript_hash = vec![];
```
Comment (duesee) [inline openmls/src/group/public_group/mod.rs:82]: This is the important change.

File: openmls/src/group/public_group/mod.rs [hunk 4]
```diff
@@ -131,9 +144,7 @@ impl PublicGroup {
 
         let group_context = GroupContext::from(group_info.clone());
 
-        let interim_transcript_hash = if group_context.epoch() == GroupEpoch::from(0) {
-            vec![]
```
Comment (duesee) [inline openmls/src/group/public_group/mod.rs:135]: This is the important change.

File: openmls/src/group/public_group/mod.rs [hunk 5]
```diff
@@ -79,7 +79,20 @@ impl PublicGroup {
...
+        let interim_transcript_hash = {
+            let input = InterimTranscriptHashInput::from(&initial_confirmation_tag);
+
+            input
+                .calculate_interim_transcript_hash(
+                    crypto,
+                    group_context.ciphersuite(),
+                    group_context.confirmed_transcript_hash(),
+                )
+                .unwrap()
```
Comment (duesee) [inline openmls/src/group/public_group/mod.rs:94]: We will need to handle this properly. As soon as we agree that this is what we want, I can make this method take a `backend` and return a `Result`.

**Review by franziskuskiefer — APPROVED** [review]
Thanks, gltm

**Pull request merged**